pkg/kubelet/cm/topologymanager/policy.go

```diff
@@ -127,7 +126,35 @@ func maxOfMinAffinityCounts(filteredHints [][]TopologyHint) int {
 	return maxOfMinCount
 }
 
-func compareHints(bestNonPreferredAffinityCount int, current *TopologyHint, candidate *TopologyHint) *TopologyHint {
+type HintMerger struct {
+	NUMAInfo                      *NUMAInfo
+	FilteredHints                 [][]TopologyHint
+	BestNonPreferredAffinityCount int
+	CompareNUMAAffinityMasks      func(bitmask.BitMask, bitmask.BitMask) bitmask.BitMask
+}
+
+func NewHintMerger(numaInfo *NUMAInfo, filteredHints [][]TopologyHint, policyName string, opts PolicyOptions) HintMerger {
```

Comment:
I'd prefer this to take `providerHints` directly and then internally calculate `filteredHints` and cache it.

Reply:
What about `single_numa` policy?
```go
func (p *singleNumaNodePolicy) Merge(providersHints []map[string][]TopologyHint) (TopologyHint, bool) {
	filteredHints := filterProvidersHints(providersHints)
	// Filter to only include don't cares and hints with a single NUMA node.
	singleNumaHints := filterSingleNumaHints(filteredHints)

	merger := NewHintMerger(p.numaInfo, singleNumaHints, PolicyBestEffort, p.opts)
```
To have custom filters in `HintMerger` we can add `CustomHintFilter` field:
```go
type HintMerger struct {
	NUMAInfo                      *NUMAInfo
	FilteredHints                 [][]TopologyHint
	BestNonPreferredAffinityCount int
	CompareNUMAAffinityMasks      func(bitmask.BitMask, bitmask.BitMask) bitmask.BitMask
        CustomHintFilter func([][]TopologyHint) [][]TopologyHint
}
```
and have it defaulted to:
```go
func DefaultHintFilter (allResourcesHints[][]TopologyHint) [][]TopologyHint{
        return allResourcesHints
}
```
or we pass filteredHints

Reply:
I see. The single numa node policy used a different filtered hints function. OK, then I guess it makes sense to pass in filtered hints as you were doing previously.